p2p/src/sync/peer.rs

```diff
@@ -92,12 +95,14 @@ where
                         expected,
                         header
                     );
-                    return Err(P2pError::InvalidData);
+                    // TODO: this is a protocol error
+                    return Err(P2pError::ProtocolError(ProtocolError::InvalidMessage));
                 }
             }
             _ => {
+                // TODO: this is a protocol error
                 log::error!("received a header from peer while not expecting it");
```

Comment:
Please try to add useful information to error messages. Mentioning the peer information, like address, is useful. If this message appears in the logs, it'd be completely useless.

Reply:
The peer ID would be logged by the PeerManager but I can log it here too